### ISSUE TITLE
Question about stores in STQ

### ISSUE THREAD

**Issue by kabylkas** [riscv-boom/riscv-boom#611]
**Type of issue**:  quick question
Hi!

I noticed that stores stay in the STQ past the commit point.

I would really appreciate it if you could shed some light on why is this the case and what is the condition that deallocates the STQ entry?

Thanks!

**Comment by ccelio** [comment]
So long as newer loads within the core can see the stores that are currently in the STQ, it is perfectly fine to hold stores in the STQ well past the commit point. Generally speaking, stores drain from the STQ when they can to free up STQ space.

Any complications caused to *other* cores are governed by the Memory Consistency Model (the legal values a load on another core may see), which is beyond the scope of a GitHub issue.

**Comment by kabylkas** [comment]
Thanks Chris! I was actually trying to understand STQ behavior in the context of multi-core. 

@ccelio, do you think it is safe to make the following assumption: **a store can be considered globally performed (visible by other cores) at the moment when the corresponding STQ entry signals `succeeded`.** Before that, it is visible only locally?

For example:

![image](https://user-images.githubusercontent.com/8511359/187762097-9c7f10f9-ca49-4dd5-88f4-d02dd7b1d30a.png)

**Comment by ccelio** [comment]
It's been too long since I've looked at this and the exact definition of the `succeeded` signal, but I believe that is correct, that signal marks when the cache has accepted the store data and it is now globally visible and the STQ entry can now be deallocated (perhaps plus or minus a cycle.. you'd want to look inside the dcache to see exactly when the store is accepted, or rather, when the cache would accept a snoop request for that line and report out the new data).

I don't know if this has been changed, but I remember boom sent stores to the cache in-order, and verified that they succeeded in-order. As I recall, regarding RISCV WMO, that is actually not a requirement.
